sdk/python/lib/pulumi/output.py

```diff
@@ -253,27 +253,32 @@ def from_input(val: Input[T]) -> 'Output[T]':
         if isinstance(val, dict):
             # Since Output.all works on lists early, serialize this dictionary into a list of lists first.
             # Once we have a output of the list of properties, we can use an apply to re-hydrate it back into a dict.
-            transformed_items = [[k, Output.from_input(v)] for k, v in val.items()]
-            return Output.all(*transformed_items).apply(lambda props: {k: v for k, v in props}, True)
+            dict_items = [[k, Output.from_input(v)] for k, v in val.items()]
+            # type checker doesn't like returing a Dict in the apply callback
+            fn = cast(Callable[[List[Any]], T], lambda props: {k: v for k, v in props}) # pylint: disable=unnecessary-comprehension
```

Comment:
I guess the issue here is that we're in a context where we "know" `T` is a `dict`, but the type checker doesn't?  

It's a shame that ends up with so much gymnastics (and having to use pretty heavyweight casts) - but perhaps theres no better way in `mypy`?

Reply:
Yup not sure there is a better way. The type checker doesn't seem to understand that Output can resolve to a Dict, so we get this error: 

`Argument 1 to "apply" of "Output" has incompatible type "Callable[[List[Any]], Dict[Any, Any]]"; expected "Callable[[List[Any]], Union[T, Awaitable[T], Output[T]]]"`